### ISSUE TITLE
Nvidia Shield Pro - Failed to authenticate 

### ISSUE THREAD

**Issue by Seexelas** [dwaan/homebridge-adb#71]
**Before you post**
Please make sure you done all the necessary steps in Prerequisite in README.md

Done, Debug Network is authorised in Debug menu. 

**Describe the situation**
Into the terminal, when I do adb connect 192.168.1.6:5555, I get an authentification error, nothing on the Tv for autorisation appears.

adb kill-server does kill the connection, but on reconnect, still failed to authenticate. 

**Log output**
pi@homebridge:/var/lib/homebridge $ adb connect 192.168.1.6:5555
failed to authenticate to 192.168.1.6:5555
pi@homebridge:/var/lib/homebridge $ adb connect 192.168.1.6:5555
already connected to 192.168.1.6:5555
pi@homebridge:/var/lib/homebridge $ adb devices
List of devices attached
192.168.1.6:5555        unauthorized

Many thanks for the help.

**Comment by dwaan** [comment]
Have you tried the `adb tcpip 5555` Before connecting?  If you done it, and it doesn’t present some error, in my experience the adb in your shirked need to be enabled using usb cable to computer first. But I’m pretty sure shield enable remote adb by default. 

You can also try using `adb kill-server` and `adb start-server` command to reset the adb server in your Homebridge.

**Comment by Seexelas** [comment]
Before upgrading to Nvidia Shield OS 9.0, it was working great. I did a hard reset and nothing works since. 

I did adb kill-server and start. 

And when I do adb tcp ip 5555

pi@homebridge:/var/lib/homebridge $ adb tcpip 5555
error: device unauthorized.
This adb server's $ADB_VENDOR_KEYS is not set
Try 'adb kill-server' if that seems wrong.
Otherwise check for a confirmation dialog on your device.

I will try with an usb cable.

**Comment by Seexelas** [comment]
I tried with ADBLINK from my PC on WiFi, and it worked without anything. I received the authorisation box on my Nvidia Shield and could connect.

From Homebridge on my Raspberry, I can not do anything. I always get authentification failure and no popup. Try on a Debian 11 VM on my Synology, I also get the same errors. Could it be a bug on the ADB released for linux ?

**Comment by Seexelas** [comment]
Hi, 

So I continued my tests, and with adb tools on Windows and on a Mac M1 on the last OS, it work perfectly. I got the prompt to authorise the device. 

I don't know if there is a bug between linux, adb tools and the nvidia shield on Android 11, but I tried every tutorials I could found, with USB or remote, I can't managed to get it working. I don't get the authorisation popup.

I did a fresh install of my raspberry, nothing changes. 

If anyone has a solution, thanks

**Comment by Pentelfrost** [comment]
You might want to check the version of ADB you have on the pi. What you're describing sounds a lot like using a version of ADB which doesn't support wireless pairing. I think you need at least 30.0.0 for that.

**Comment by xologram** [comment]
I had the same problem on pi with Raspbian 11 and adb that comes with it. Raspbian 10 worked fine so I ended up installing Homebridge on 10.

**Comment by Flauchhaus** [comment]
I had the same issue on a homebridge instance installed on a debian 11 container on proxbox.

The solution was to use the most recent adb version from google directly instead of installing the (outdated) adb version that is provided by the debian repository. It showed the unauthorized error the first time but it did finally prompt on the shield to accept the connection, afterwards it always connected successfully.

Short version:
download latest tools from https://developer.android.com/studio/releases/platform-tools instead of installing it with apt.
Extract it, then move just the adb application to /usr/bin/ and add executable rights.

**Comment by sambal0** [comment]
Hello @Flauchhaus,
Tried your solution and get the answer "-bash: /usr/bin/adb: cannot execute binary file: Exec format error".
Do you have an idea to solve this issue?

**Comment by Flauchhaus** [comment]
@sambal0 It is likely that the adb binary from the android developer page is x86 only, it wont run on arm based chips. You'd need to get an updated adb binary from somewhere else I'm afraid.

**Comment by sambal0** [comment]
Thanks for your reply, @Flauchhaus.
Actually I´ve read somewhere that the current platform-tools needs a 64 bit OS so I already updated my Raspberry from 32 to 64. This update didn´t solve the problem. That´s why I tried it with the request for help here. 
Do you know, if a adb binary version above 30.0.0 exists that´ll work with my configuration? If yes, where can I find it? Thanks a lot in advance

Raspberry 3B: Raspbian GNU/Linux Bullseye (11) / aarch64
Homebridge: 1.5.0
Node: 16.17.0
Npm: 8.18.0

**Comment by d3smo** [comment]
> adb tcpip 5555



> Hi,
> 
> So I continued my tests, and with adb tools on Windows and on a Mac M1 on the last OS, it work perfectly. I got the prompt to authorise the device.
> 
> I don't know if there is a bug between linux, adb tools and the nvidia shield on Android 11, but I tried every tutorials I could found, with USB or remote, I can't managed to get it working. I don't get the authorisation popup.
> 
> I did a fresh install of my raspberry, nothing changes.
> 
> If anyone has a solution, thanks

Hi Seexelas, I'm experimenting exactly the same issue to send adb command form my Raspberry (running Raspbian) to My Nvidia Shield,. No prompting windows and device is unauthorized even after refreshing everything, do you find an issue to your issue that looks like mine ?

**Comment by secprentice** [comment]
> > adb tcpip 5555
> 
> > Hi,
> > So I continued my tests, and with adb tools on Windows and on a Mac M1 on the last OS, it work perfectly. I got the prompt to authorise the device.
> > I don't know if there is a bug between linux, adb tools and the nvidia shield on Android 11, but I tried every tutorials I could found, with USB or remote, I can't managed to get it working. I don't get the authorisation popup.
> > I did a fresh install of my raspberry, nothing changes.
> > If anyone has a solution, thanks
> 
> Hi Seexelas, I'm experimenting exactly the same issue to send adb command form my Raspberry (running Raspbian) to My Nvidia Shield,. No prompting windows and device is unauthorized even after refreshing everything, do you find an issue to your issue that looks like mine ?

Your issue is being caused by the version of adb your using is out of date and won't auth with your Droid TV. We need to find the newer version of ADB for Arm processors like in the Pi .
